packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmUtils.kt

```diff
@@ -196,14 +188,19 @@ internal fun <T : BaseRealmObject> copyToRealm(
         }
         val target = if (hasPrimaryKey) {
             inputScope {
-                create(
-                    mediator,
-                    realmReference,
-                    element::class,
-                    className,
-                    convertArg(primaryKey),
-                    updatePolicy
-                )
+                try {
+                    create(
+                        mediator,
+                        realmReference,
+                        element::class,
+                        className,
+                        convertArg(primaryKey),
+                        updatePolicy
+                    )
+                } catch (e: IllegalStateException) {
+                    // TODO https://github.com/realm/realm-core/issues/6223
```

Comment:
This was closed as a WontFix. I see arguments for both cases, but just to avoid breaking changes, keeping the mapping here is probably slightly better. So I would just remove this TODO, but maybe explain the rationale instead.